app/assets/javascripts/admin/addon/components/themes-grid-placeholder.gjs

```diff
@@ -28,59 +28,325 @@ export default class ThemesGridPlaceholder extends Component {
     }
   }
 
+  get safeThemeColors() {
+    let colors = this.themeColors;
+
+    return {
+      primary: htmlSafe(colors.primary),
+      secondary: htmlSafe(colors.secondary),
+      tertiary: htmlSafe(colors.tertiary),
+      quaternary: htmlSafe(colors.quaternary),
+      highlight: htmlSafe(colors.highlight),
+      danger: htmlSafe(colors.danger),
+      success: htmlSafe(colors.success),
+      love: htmlSafe(colors.love),
+    };
+  }
+
+  get randomVariant() {
+    return (this._variant ||= Math.floor(Math.random() * 4) + 1);
+  }
```

Comment:
how about making this a regular property calculated when the component is created? (the line will need to be moved to the top of the class)

```suggestion
  randomVariant = Math.floor(Math.random() * 4) + 1);
```